include/envoy/http/codec.h

```diff
@@ -150,11 +150,17 @@ class ConnectionCallbacks {
 /**
  * A list of options that can be specified when creating a codec.
  */
-class CodecOptions {
+class CodecOptionFlags {
```

Comment:
This whole thing is a mess and will be cleaned up as part of the v2 API effort. cc @htuch @PiotrSikora 

In the interim, I think it's OK to leave this in the http namespace.